test/e2eprovider/server/server.go

```diff
@@ -0,0 +1,144 @@
+
+	"google.golang.org/grpc"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/yaml"
+)
+
+// Server is a mock csi-provider server
+type Server struct {
+	grpcServer *grpc.Server
+	listener   net.Listener
+	SocketPath string
+	network    string
+	Vault      vault.Vault
+}
+
+// NewE2eProviderServer returns a mock csi-provider grpc server
+func NewE2eProviderServer(endpoint string, vault vault.Vault) (*Server, error) {
+	network, address, err := util.ParseEndpoint(endpoint)
+	if err != nil {
+		klog.Fatal(err.Error())
+	}
+
+	server := grpc.NewServer()
+	s := &Server{
+		grpcServer: server,
+		SocketPath: address,
+		network:    network,
+		Vault:      vault,
+	}
+
+	v1alpha1.RegisterCSIDriverProviderServer(server, s)
+
+	return s, nil
+}
+
+// Start starts the mock csi-provider server
+func (m *Server) Start() error {
+	var err error
+
+	m.listener, err = net.Listen(m.network, m.SocketPath)
+	if err != nil {
+		return err
+	}
+
+	klog.Infof("Listening for connections on address: %v", m.listener.Addr())
+	go m.grpcServer.Serve(m.listener)
+	return nil
+}
+
+// Stop stops the mock csi-provider server
+func (m *Server) Stop() {
+	m.grpcServer.GracefulStop()
+}
+
+// Mount implements provider csi-provider method
+func (m *Server) Mount(ctx context.Context, req *v1alpha1.MountRequest) (*v1alpha1.MountResponse, error) {
+	var attrib, secret map[string]string
+	var filePermission os.FileMode
+	var err error
+
+	resp := &v1alpha1.MountResponse{
+		ObjectVersion: []*v1alpha1.ObjectVersion{},
+	}
+
+	if err = json.Unmarshal([]byte(req.GetAttributes()), &attrib); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal attributes, error: %+v", err)
+	}
+	if err = json.Unmarshal([]byte(req.GetSecrets()), &secret); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal secrets, error: %+v", err)
+	}
+	if err = json.Unmarshal([]byte(req.GetPermission()), &filePermission); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal file permission, error: %+v", err)
+	}
+	if len(req.GetTargetPath()) == 0 {
+		return nil, fmt.Errorf("missing target path")
+	}
+
+	objectsStrings := attrib["objects"]
+	if objectsStrings == "" {
+		return nil, fmt.Errorf("objects is not set")
+	}
+
+	var objects types.StringArray
+	err = yaml.Unmarshal([]byte(objectsStrings), &objects)
+	if err != nil {
+		return nil, fmt.Errorf("failed to yaml unmarshal objects, error: %w", err)
+	}
+
+	keyVaultObjects := []types.KeyVaultObject{}
+	for i, object := range objects.Array {
+		var keyVaultObject types.KeyVaultObject
+		err = yaml.Unmarshal([]byte(object), &keyVaultObject)
+		if err != nil {
+			return nil, fmt.Errorf("unmarshal failed for keyVaultObjects at index %d, error: %w", i, err)
+		}
+
+		keyVaultObjects = append(keyVaultObjects, keyVaultObject)
+	}
+
+	for _, keyVaultObject := range keyVaultObjects {
+		fileName := keyVaultObject.ObjectName
+		if keyVaultObject.ObjectAlias != "" {
+			fileName = keyVaultObject.ObjectAlias
```

Comment:
These are specific to azure provider and I don't think we need it here. We should have a simple logic for the RPC calls without too many validations as it's more to test the driver logic than provider. WDYT?

Reply:
We need this in order to test `alias`.

Reply:
Removed alias